cmd/hypper/status.go

```diff
@@ -0,0 +1,177 @@
+/*
+Copyright The Helm Authors, SUSE
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"github.com/Masterminds/log-go"
+	logio "github.com/Masterminds/log-go/io"
+	"github.com/rancher-sandbox/hypper/pkg/action"
+	"github.com/spf13/cobra"
+	"helm.sh/helm/v3/cmd/helm/require"
+	"helm.sh/helm/v3/pkg/chartutil"
+	"helm.sh/helm/v3/pkg/cli/output"
+	"helm.sh/helm/v3/pkg/release"
+	"io"
+	"strings"
+	"time"
+)
+
+// NOTE: Keep the list of statuses up-to-date with pkg/release/status.go.
+var statusHelp = `
+This command shows the status of a named release.
+The status consists of:
+- last deployment time
+- k8s namespace in which the release lives
+- state of the release (can be: unknown, deployed, uninstalled, superseded, failed, uninstalling, pending-install, pending-upgrade or pending-rollback)
+- revision of the release
+- description of the release (can be completion message or error message, need to enable --show-desc)
+- list of resources that this release consists of, sorted by kind
+- details on last test suite run, if applicable
+- additional notes provided by the chart
+`
+
+func newStatusCmd(cfg *action.Configuration, logger log.Logger) *cobra.Command {
+	client := action.NewStatus(cfg)
+	var outfmt output.Format
+
+	cmd := &cobra.Command{
+		Use:   "status RELEASE_NAME",
+		Short: "display the status of the named release",
+		Long:  statusHelp,
+		Args:  require.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Get an io.Writer compliant logger instance at the info level.
+			wInfo := logio.NewWriter(logger, log.InfoLevel)
+
+			rel, err := client.Run(args[0])
+			if err != nil {
+				return err
+			}
+
+			// strip chart metadata from the output
+			rel.Chart = nil
+
+			return outfmt.Write(wInfo, &statusPrinter{rel, false, client.ShowDescription})
+		},
+	}
+
+	f := cmd.Flags()
+
```

Comment:
nitpick: we could add the `--revision` and `--description` flags here, as Helm does

Reply:
great point, I missed that flag!

Reply:
should be fixed now!